storage/txcache/evictionStrategy.go

```diff
@@ -0,0 +1,168 @@
+package txcache
+
+import (
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/logger"
+)
+
+var log = logger.GetOrCreate("txcache/eviction")
+
+// EvictionStrategyConfig is a cache eviction model
+type EvictionStrategyConfig struct {
+	CountThreshold                 int
+	CountJustAFewSenders           int
+	NoOldestSendersToEvict         int
+	ALotOfTransactionsForASender   int
+	NoTxsToEvictForASenderWithALot int
+}
+
+// EvictionStrategy is a cache eviction model
+type EvictionStrategy struct {
+	Cache  *TxCache
+	Config EvictionStrategyConfig
+	mutex  sync.Mutex
+}
+
+// NewEvictionStrategy creates a new EvictionModel
+func NewEvictionStrategy(cache *TxCache, config EvictionStrategyConfig) *EvictionStrategy {
+	model := &EvictionStrategy{
+		Cache:  cache,
```

Comment:
you need to check if cache is nil or config is wrong and return error in those cases.

Reply:
Added `NewTxCacheWithEviction`, and it receives the config struct now (no nil).